compiler/tests/fixtures/demos/uniswap.fe

```diff
@@ -0,0 +1,359 @@
+contract UniswapV2Factory:
+    fee_to: address
+    fee_to_setter: address
+
+    pairs: map<address, map<address, address>>
+
+    # todo: discuss unbounded storage arrays
+    all_pairs: address[0]
+
+    event PairCreated:
+        idx token0: address
+        idx token1: address
+        pair: address
+        index: u256
+
+    pub def __init__(fee_to_setter: address):
+        self.fee_to_setter = fee_to_setter
+
+    pub def fee_to() -> address:
+        return self.fee_to
+
+    pub def fee_to_setter() -> address:
+        return self.fee_to_setter
+
+    pub def all_pairs_length() -> u256:
+        return self.all_pairs.length
+
+    pub def create_pair(token_a: address, token_b: address) -> address:
+        assert token_a != token_b, "UniswapV2: IDENTICAL_ADDRESSES"
+
+        # (token0: address, token1: address) = (tokenA, tokenB) if tokenA < tokenB else (tokenB, tokenA)
+        assert token0 != address(0), "UniswapV2: ZERO_ADDRESS"
+        assert pairs[token0][token1] == address(0), "UniswapV2: PAIR_EXISTS"
+
+        salt: u256 = (token0, token1).abiEncodePacked().keccak256()
+        pair: UniswapV2Pair = UniswapV2Pair.create2(0, salt)
+        pair.initialize(token0, token1)
+
+        self.pairs[token0][token1] = pair.address
+        self.pairs[token1][token0] = pair.address
+        self.all_pairs.push(pair.address)
+
+        emit PairCreated(token0, token1, pair.address, all_pairs.length)
+        return pair.address
+
+    pub def set_fee_to(fee_to: address):
+        assert msg.sender == self.fee_to_setter, "UniswapV2: FORBIDDEN"
+        self.fee_to = fee_to
+
+    pub def set_fee_to_setter(fee_to_setter: address):
+        assert msg.sender == fee_to_setter, "UniswapV2: FORBIDDEN"
+        self.fee_to_setter = fee_to_setter
+
+
+# This needs to be merged with UniswapV2Pair, since we do not support inheritance yet.
+# For the sake of readability though, we can leave it separate until we're ready to
+# deploy and test.
+contract UniswapV2ERC20:
+    # `const` is not supported by the parser yet.
+    # const name: string10 = "Uniswap V2"
+    # const symbol: string6 = "UNI-V2"
```

Comment:
This seems like a low hanging fruit that we could plan for soon. We should create an issue for it.